pkg/api/types.go

```diff
@@ -1107,9 +1107,12 @@ type NodeSpec struct {
 	// Note: assigning IP ranges to nodes might need to be revisited when we support migratable IPs.
 	PodCIDR string `json:"podCIDR,omitempty"`
 
-	// External ID of the node assigned by some machine database (e.g. a cloud provider)
+	// External ID of the node assigned by some machine database (e.g. a cloud provider) (obsolete)
 	ExternalID string `json:"externalID,omitempty"`
 
+	// ID of the cloud provider that is exposing this machine
```

Comment:
The way this comment is written, I'd expect it to be filled in with "gce" or "aws". But the field it's replacing could be used along with the cloud provider to actually identify the node. I think it's fine to have a cloud provider as a field in the node spec, but I think we also need an instanceID that uniquely identifies this particular node within the cloud provider. 

If this is meant to be the unique identifer, then the comment is confusing, as is the similarity with the function in the cloud provider interface below.